include/swift/AST/DiagnosticConsumer.h

```diff
@@ -188,8 +188,9 @@ class FileSpecificDiagnosticConsumer : public DiagnosticConsumer {
   /// None if diagnostic is to be distributed to every consumer,
   /// a particular consumer if diagnostic goes there.
   ///
-  /// (For Xcode compatibility, batch mode must suppress diagnostics to
-  /// non-primaries.)
+  /// (If batch mode failed to suppress diagnostics for non-primary files, it
+  /// would end up writing each of those diagnostics in each primary file's
+  /// serialized diagnostic file.)
```

Comment:
Sorry, I should have been clearer, because this is still describing the behavior (which is already on the line above). I don't think you should have any comment here at all, because it's not FileSpecificDiagnosticConsumer's job to decide why some diagnostics are suppressed. If you have a comment at the client side, it should note that we're relying on diagnostics located in a non-primary file already appearing when you compile that file.

Reply:
Thanks, that's easy: Comment deleted.